ui/tests/acceptance/secrets/backend/kv/kv-v2-workflow-create-test.js

```diff
@@ -38,101 +30,1067 @@ module('Acceptance | kv-v2 workflow | secret and version create', function (hook
 
   module('admin persona', function (hooks) {
     hooks.beforeEach(async function () {
-      const token = await runCmd([
-        createPolicyCmd('admin', adminPolicy(this.backend)),
-        createTokenCmd('admin'),
-      ]);
+      const token = await runCmd(tokenWithPolicyCmd('admin', personas.admin(this.backend)));
       await authPage.login(token);
     });
-    test.skip('create new root secret flow', async function (assert) {
-      assert.expect(0);
-      // TODO: navigate to kv backend from backends
-      // ✅ shows list items / no empty state
-      // ✅ shows correct toolbar options
-      // click "Create secret"
-      // ✅ URL is vault/secrets/:backend/kv/create
-      // ✅ Path input is empty
-      // click [data-test-kv-save]
-      // ✅ Shows validation errors
+    test('cancel on create clears model (a)', async function (assert) {
+      const backend = this.backend;
+      await visit(`/vault/secrets/${backend}/kv/list`);
+      assert.dom(PAGE.list.item()).exists({ count: 1 }, 'single secret exists on list');
+      assert.dom(PAGE.list.item('app/')).hasText('app/', 'expected list item');
+      await click(PAGE.list.createSecret);
+      await fillIn(FORM.inputByAttr('path'), 'jk');
+      await click(FORM.cancelBtn);
+      assert.dom(PAGE.list.item()).exists({ count: 1 }, 'same amount of secrets');
+      assert.dom(PAGE.list.item('app/')).hasText('app/', 'expected list item');
+      await click(PAGE.list.createSecret);
+      await fillIn(FORM.inputByAttr('path'), 'psych');
+      await click(PAGE.breadcrumbAtIdx(1));
+      assert.dom(PAGE.list.item()).exists({ count: 1 }, 'same amount of secrets');
+      assert.dom(PAGE.list.item('app/')).hasText('app/', 'expected list item');
+    });
+    test('cancel on new version rolls back model (a)', async function (assert) {
+      const backend = this.backend;
+      await visit(`/vault/secrets/${backend}/kv/${encodeURIComponent('app/first')}/details`);
+      assert.dom(PAGE.infoRowValue('foo')).exists('key has expected value');
+      await click(PAGE.detail.createNewVersion);
+      await fillIn(FORM.keyInput(), 'bar');
+      await click(FORM.cancelBtn);
+      assert.dom(PAGE.infoRowValue('foo')).exists('key rolls back to previous value');
```

Comment:
Since we're using `createRecord` each time and not finding the record and updating - I'm not sure checking for rolling back model attributes is doing what we think it does?

Maybe instead we should check that the record has been unloaded?

Reply:
Well, my thinking was that the workflow tests should mostly test that the user experience is what we expect. But I've come up against a few cases where I wondered if we should check the API payload, etc. I didn't add all that because it would make the tests even longer, but do you think it's worth adding here?

Reply:
yeah - that makes sense. I guess seeing `rolls back` is a little confusing since I'd expect to see a `findRecord` for the model in the code. 

🤔  Hmm, that's a good question. I think testing that UI interactions yield what we expect is sufficient. Testing payloads are submitted correctly can be reserved for component tests - like we did [here](https://github.com/hashicorp/vault/blob/8bd976b320ef98b81cc66f391b7008578d044ee1/ui/tests/integration/components/kv/page/kv-page-secrets-create-test.js#L44-L50)